src/main/java/co/inventorsoft/springmail/dao/EmailDAO.java

```diff
@@ -0,0 +1,103 @@
+package co.inventorsoft.springmail.dao;
+
+import co.inventorsoft.springmail.model.Email;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.FileSystemUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Repository
+public class EmailDAO {
+    private String repLocation;
+
+    @Autowired
+    public EmailDAO(Environment environment) {
+        this.repLocation = Objects.requireNonNull(FileSystemUtils.class.getClassLoader().getResource(""))
+                .getPath() + environment.getProperty("emails.storage.location");
+    }
+
+    public List<Email> findAll() throws IOException {
+        List<Email> emailsList = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
please, declare ObjectMapper as bean in separate configuration and inject it as dependency